lib/gui-utils/src/GUIUtils.cpp

```diff
@@ -93,103 +118,292 @@ void GUIUtils::displayEndLine(String msg) {
   u8g2.print(msg.c_str());
 }
 
-void GUIUtils::displaySensorAvarage(int avarage) {
+void GUIUtils::displaySensorAvarage(int avarage)
+{
+#ifndef Emoticons
+#ifdef TTGO_TQ
+  if (avarage < 13)
+  {
+    u8g2.setFont(u8g2_font_unifont_t_emoticons);
+    u8g2.drawGlyph(76, 12, 0x0024);
+    u8g2.setFont(u8g2_font_5x7_tf);
+    u8g2.setCursor(77, 16);
+    u8g2.print("GOOD");
+  }
+  else if (avarage < 36)
+  {
+    u8g2.setFont(u8g2_font_unifont_t_emoticons);
+    u8g2.drawGlyph(76, 12, 0x0062);
+    u8g2.setFont(u8g2_font_4x6_tf);
+    u8g2.setCursor(77, 17);
+    u8g2.print("MODERATE");
+  }
+  else if (avarage < 56)
+  {
+    u8g2.setFont(u8g2_font_unifont_t_emoticons);
+    u8g2.drawGlyph(76, 12, 0x0032);
+    u8g2.setFont(u8g2_font_4x6_tf);
+    u8g2.setCursor(77, 17);
+    u8g2.print("UNHEALT S");
+  }
+  else if (avarage < 151)
+  {
+    u8g2.setFont(u8g2_font_unifont_t_emoticons);
+    u8g2.drawGlyph(76, 12, 0x0051);
+    u8g2.setFont(u8g2_font_4x6_tf);
+    u8g2.setCursor(77, 17);
+    u8g2.print("UNHEALTY");
+  }
+  else if (avarage < 251)
+  {
+    u8g2.setFont(u8g2_font_unifont_t_emoticons);
+    u8g2.drawGlyph(76, 12, 0x0053);
+    u8g2.setFont(u8g2_font_4x6_tf);
+    u8g2.setCursor(77, 17);
+    u8g2.print("VERY UNH");
+  }
+  else
+  {
+    u8g2.setFont(u8g2_font_unifont_t_emoticons);
+    u8g2.drawGlyph(76, 12, 0x0057);
+    u8g2.setFont(u8g2_font_4x6_tf);
+    u8g2.setCursor(77, 17);
+    u8g2.print("HAZARDOUS");
+  }
+#else
+  delay(1);
+#endif
+#else
+  if (avarage < 13)
+  {
+#ifdef TTGO_TQ
+    u8g2.drawXBM(1, 0, 32, 32, Smileface1);
+    u8g2.setFont(u8g2_font_6x12_tf);
+    u8g2.setCursor(40, 15);
+    u8g2.print("BUENO/");
+    u8g2.setFont(u8g2_font_5x8_tf);
+    u8g2.setCursor(76, 16);
+    u8g2.print("verde");
+#else
+    u8g2.drawXBM(0, 1, 32, 32, Smileface1);
+    u8g2.setFont(u8g2_font_6x12_tf);
+    u8g2.setCursor(40, 15);
+    u8g2.print("BUENO");
+#endif
+  }
+  else if (avarage < 36)
+  {
+#ifdef TTGO_TQ
+    u8g2.drawXBM(1, 0, 32, 32, Smileface2);
+    u8g2.setFont(u8g2_font_6x12_tf);
+    u8g2.setCursor(38, 15);
+    u8g2.print("MODERADO/");
+    u8g2.setFont(u8g2_font_6x10_tf);
+    u8g2.setCursor(92, 16);
+    u8g2.print("ama");
+#else
+    u8g2.drawXBM(0, 1, 32, 32, Smileface2);
+    u8g2.setFont(u8g2_font_5x7_tf);
+    u8g2.setCursor(25, 29);
+    u8g2.print("MODERADO");
+#endif
```

Comment:
please specialize it in two methods for not have repeated code.

Reply:
Resolved in branch:
https://github.com/danielbernalb/esp32-hpma115s0/tree/dv/devel_emoticons2